src/swap_utils.c

```diff
@@ -38,6 +43,11 @@ bool parse_swap_config(const uint8_t *config, uint8_t config_len, char *ticker,
     if (config_len - offset < 1) {
         return false;
     }
-    *decimals = config[offset];
+    *decimals = config[offset++];
+    // optional for retro-compatibility
+    if ((config_len - offset) >= sizeof(*chain_id)) {
```

Comment:
I think we can differentiate 3 cases:
- legacy size
- legacy size + sizeof(*chain_id)
- something else => error

Reply:
I would like it to ignore the leftover data if there is any (just like it was up until now), so we can keep on adding things at the end if the need ever arises.